Build failure (can't find Xrandr.h)

From a `Pkg.update()` this morning:

```
INFO: Building GLFW
-- The C compiler identification is GNU 4.8.2
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - done
-- Found X11: /usr/lib/x86_64-linux-gnu/libX11.so
-- Found OpenGL: /usr/lib/x86_64-linux-gnu/libGL.so  
-- Looking for pthread_create
-- Looking for pthread_create - not found
-- Using X11 for window creation
-- Using GLX for context creation
-- Configuring done
CMake Warning (dev) in src/CMakeLists.txt:
  Policy CMP0022 is not set: INTERFACE_LINK_LIBRARIES defines the link
  interface.  Run "cmake --help-policy CMP0022" for policy details.  Use the
  cmake_policy command to set the policy and suppress this warning.

  Target "glfw" has an INTERFACE_LINK_LIBRARIES property which differs from
  its LINK_INTERFACE_LIBRARIES properties.

  INTERFACE_LINK_LIBRARIES:

    /usr/lib/x86_64-linux-gnu/libX11.so;-lpthread;/usr/lib/x86_64-linux-gnu/librt.so;/usr/lib/x86_64-linux-gnu/libXrandr.so;/usr/lib/x86_64-linux-gnu/libXi.so;/usr/lib/x86_64-linux-gnu/libXxf86vm.so;/usr/lib/x86_64-linux-gnu/librt.so;/usr/lib/x86_64-linux-gnu/libm.so;/usr/lib/x86_64-linux-gnu/libGL.so

  LINK_INTERFACE_LIBRARIES:



This warning is for project developers.  Use -Wno-dev to suppress it.

-- Generating done
-- Build files have been written to: /home/tim/.julia/v0.3/GLFW/deps/builds/glfw-3.0.4
Scanning dependencies of target glfw
[  5%] Building C object src/CMakeFiles/glfw.dir/clipboard.c.o
In file included from /home/tim/.julia/v0.3/GLFW/deps/src/glfw-3.0.4/src/internal.h:73:0,
                 from /home/tim/.julia/v0.3/GLFW/deps/src/glfw-3.0.4/src/clipboard.c:27:
/home/tim/.julia/v0.3/GLFW/deps/src/glfw-3.0.4/src/x11_platform.h:42:35: fatal error: X11/extensions/Xrandr.h: No such file or directory
 #include <X11/extensions/Xrandr.h>
                                   ^
compilation terminated.
make[2]: *** [src/CMakeFiles/glfw.dir/clipboard.c.o] Error 1
make[1]: *** [src/CMakeFiles/glfw.dir/all] Error 2
make: *** [all] Error 2
=====================================================================[ ERROR: GLFW ]=====================================================================

failed process: Process(`make install`, ProcessExited(2)) [2]
while loading /home/tim/.julia/v0.3/GLFW/deps/build.jl, in expression starting on line 16

=========================================================================================================================================================

====================================================================[ BUILD ERRORS ]=====================================================================

WARNING: GLFW had build errors.

 - packages with build errors remain installed in /home/tim/.julia/v0.3
 - build a package and all its dependencies with `Pkg.build(pkg)`
 - build a single package by running its `deps/build.jl` script

=========================================================================================================================================================
```

For me, `locate Xrandr.h` turns up nothing. I'm on Kubuntu 14.04.

Also note the `Target "glfw" has an INTERFACE_LINK_LIBRARIES property which differs from
  its LINK_INTERFACE_LIBRARIES properties` warning.

--- a/deps/build.jl
+++ b/deps/build.jl
@@ -2,6 +2,8 @@
 const version = VersionNumber(3, 0, 4)
 const glfw = "glfw-$version"
 
+# TODO: if the latest version is already installed, don't bother with any of this
+
 # download and compile the library from source
 @unix_only begin
 	mkpath("downloads")
@@ -13,16 +15,22 @@ const glfw = "glfw-$version"
 	mkpath("src")
 	run(`tar xzf $tarball -C src`)
 	map(mkpath, ("builds/$glfw", "usr$WORD_SIZE"))
-	cd("builds/$glfw") do
-		options = map(x -> "-D$(x[1])=$(x[2])", [
-			"BUILD_SHARED_LIBS"     => "ON",
-			"CMAKE_INSTALL_PREFIX"  => "../../usr$WORD_SIZE",
-			"GLFW_BUILD_DOCS"       => "OFF",
-			"GLFW_BUILD_EXAMPLES"   => "OFF",
-			"GLFW_BUILD_TESTS"      => "OFF"
-		])
-		run(`cmake $options ../../src/$glfw`)
-		run(`make install`)
+	try
+		info("Building GLFW $version library from source")
+		cd("builds/$glfw") do
+			options = map(x -> "-D$(x[1])=$(x[2])", [
+				"BUILD_SHARED_LIBS"     => "ON",
+				"CMAKE_INSTALL_PREFIX"  => "../../usr$WORD_SIZE",
+				"GLFW_BUILD_DOCS"       => "OFF",
+				"GLFW_BUILD_EXAMPLES"   => "OFF",
+				"GLFW_BUILD_TESTS"      => "OFF"
+			])
+			run(`cmake $options ../../src/$glfw`)
+			run(`make install`)
+		end
+	catch e
+		warn(e)
+		warn("Build failed, you may need to manually install the library (see http://www.glfw.org/download.html for more information)")
 	end
 end
 

